terraform/errors.go

```diff
@@ -0,0 +1,35 @@
+package terraform
+
+type CustomError struct {
```

Comment:
Appreciate this custom Error type, we might extend it in the future to provide extra diagnostic information as we progress in implementing the HCL parsing and JSON serialisation.